### PR TITLE
chore: Refactor docker-compose to simplify installation experience

### PR DESCRIPTION
**Pull request by ppawlowski**
## Description

This pull request refactors the 'main' docker compose file used for FlowFuse installation on self-hosted environments. Additionally, it introduces:
* `.env.example` file for persistent configuration
* `docker-compose-tls.override.yml` file for TLS management

Included changes should be merged together with https://github.com/FlowFuse/flowfuse/pull/4665

## Related Issue(s)

https://github.com/FlowFuse/flowfuse/issues/4633

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -6,27 +95,21 @@ services:
     restart: always
     volumes:
       - "/var/run/docker.sock:/tmp/docker.sock:ro"
-      - "./nginx/vhost.d:/etc/nginx/vhost.d"
```
Comment (hardillb) [inline docker-compose.yml:9]: I added this for a specific reason, I think it was to be able to overwrite defaults for specific hosts like max payload size

**Review by hardillb — COMMENTED** [review]
The upgrade instructions are going to be interesting with the changes to named volumes over local directories.

How will an exisiting user upgrade and keep using the local directories or copy them to the named volumes?

File: docker-compose.yml (hunk 1 above)
Comment (ppawlowski) [reply docker-compose.yml:9]: Custom nginx config is defined [here](https://github.com/FlowFuse/docker-compose/pull/160/files#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3R67) and attached to the container as a [config](https://github.com/FlowFuse/docker-compose/pull/160/files#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3R101).
During testing the new compose file I couldn't find any reason of mounting this directory (which is empty by default) to the nginx container. Without this binding whole setup worked without issues. That is why I decided to remove this volume binding.

**Comment by ppawlowski** [comment]
> The upgrade instructions are going to be interesting with the changes to named volumes over local directories.
> 
> How will an exisiting user upgrade and keep using the local directories or copy them to the named volumes?

I added [upgrade instructions](https://github.com/FlowFuse/docker-compose/pull/160/files#diff-434625111321abb225365806085a3753cb854d098ed9b4b6dc2001235c7f7bc4) which I tested beforehand. Feel free to verify them on your end as well.

**Review by hardillb — APPROVED** [review]
Looks good.

Only question, where is TLS_ENABLED set? I only see it mentioned in the `docker-compose.yml` but not anywhere else

**Comment by ppawlowski** [comment]
> Only question, where is TLS_ENABLED set? I only see it mentioned in the `docker-compose.yml` but not anywhere else

Added to `.env.example` to avoid confusion.

**Comment by ppawlowski** [comment]
@hardillb I've extended compose file and added the possibility to configure smtp settings. Please review one more time.

**Review by hardillb — APPROVED** [review]

**Pull request merged**